### PR TITLE
Fix bug where telemetry failed to connect when preference was changed after RC connected

### PR DESCRIPTION
**Pull request by ryandoherty**

### CODE REVIEW

File: main.py [hunk 1]
```diff
@@ -92,6 +92,8 @@ class RaceCaptureApp(App):
 
     _status_pump = StatusPump()
 
+    _rc_connected = False
```
Comment (brentpicasso) [inline main.py]: hm, is there no other good way to know if we're connected or not?

File: main.py [hunk 2]
```diff
@@ -404,6 +406,7 @@ def rc_detect_win(self, version):
             self.showStatus("{} v{}.{}.{}".format(version.friendlyName, version.major, version.minor, version.bugfix), False)
             self._data_bus_pump.start(self._databus, self._rc_api, self._rc_api.comms.supports_streaming)
             self._status_pump.start(self._rc_api)
+            self._rc_connected = True
```
Comment (brentpicasso) [inline main.py]: could we put 

self._telemetry_connection.data_connected = True

here instead?

File: main.py (hunk 1 above)
Comment (ryandoherty) [inline main.py]: We have a few other variables tracking some information we get from RCP, but nothing explicit. Best we have is if the config is loaded, but that sticks around after a disconnect. I've dug around, can't seem to find anything that works, but if there is I'm open to using it.

File: main.py (hunk 2 above)
Comment (ryandoherty) [inline main.py]: Maybe, will test.

File: main.py [hunk 3]
```diff
@@ -490,6 +491,7 @@ def _on_preferences_change(self, menu, config, section, key, value):
                 if self.rc_config.connectivityConfig.cellConfig.cellEnabled:
                     alertPopup('Telemetry error', "Turn off RaceCapture's telemetry module for app to stream telemetry.")
                 self._telemetry_connection.telemetry_enabled = True
+                self._telemetry_connection.data_connected = True
```
Comment (brentpicasso) [inline main.py]: Hm, does this do anything weird or is it harmless?

Comment (ryandoherty) [inline main.py]: It works fine from my testing, although it does seem wrong to do it on prefs change because RCP may not have connected yet.

Comment (ryandoherty) [inline main.py]: This is why I added a `_rc_connected` variable to track if the app is connected, then it's pretty explicit.

Comment (brentpicasso) [inline main.py]: right.  So is this line needed at all? since we'll be setting it above once we've connected?

Comment (ryandoherty) [inline main.py]: Nope, it's not, fixing.

**Comment by brentpicasso** [comment]
:shipit:

**Pull request merged**